File: app/src/main/java/com/example/oderfoodapp/ViewHolder/OrderViewHolder.java
package com.example.oderfoodapp.ViewHolder;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.oderfoodapp.Interface.ItemClickListener;
import com.example.oderfoodapp.R;

public class OrderViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
    public TextView txtId,txtStatus,txtPhone,txtAddress;
    private ItemClickListener itemClickListener;

    public OrderViewHolder(@NonNull View itemView) {
        super(itemView);
        txtId = itemView.findViewById(R.id.order_id);
        txtStatus = itemView.findViewById(R.id.order_status);
        txtPhone = itemView.findViewById(R.id.order_phone);
        txtAddress   = itemView.findViewById(R.id.order_address);

        itemView.setOnClickListener(this);

    }

    public void setItemClickListener(ItemClickListener itemClickListener) {
        this.itemClickListener = itemClickListener;
    }

    @Override
    public void onClick(View view) {
        //itemClickListener.onClick(view,getAdapterPosition(),false);
    }
}

File: app/src/main/java/com/example/oderfoodapp/view/view_holder/OrderViewHolder.kt
package com.example.oderfoodapp.view.view_holder

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.oderfoodapp.`interface`.ItemClickListener
import com.example.oderfoodapp.R

class OrderViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
    var txtId: TextView
    var txtStatus: TextView
    var txtPhone: TextView
    var txtAddress: TextView
    private var itemClickListener: ItemClickListener? = null

    init {
        txtId = itemView.findViewById(R.id.order_id)
        txtStatus = itemView.findViewById(R.id.order_status)
        txtPhone = itemView.findViewById(R.id.order_phone)
        txtAddress = itemView.findViewById(R.id.order_address)
        itemView.setOnClickListener(this)
    }

    fun setItemClickListener(itemClickListener: ItemClickListener?) {
        this.itemClickListener = itemClickListener
    }

    override fun onClick(view: View) {
        //itemClickListener.onClick(view,getAdapterPosition(),false);
    }
}